csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs

```diff
@@ -687,6 +687,33 @@ public IDisposableReadOnlyCollection<DisposableNamedOnnxValue> Run(IReadOnlyColl
             }
         }
 
+        /// <summary>
+        /// Ends profiling for the session. Returns the profile file name.
+        /// 
+        public string EndProfiling()
+        {
+            IntPtr nameHandle = IntPtr.Zero;
+            string str = null;
+
+            IntPtr status = NativeMethods.OrtSessionEndProfiling(_nativeHandle, 
```

Comment:
Why make this call outside of the 'try'?

Also why no 'catch' with handling or logging of exceptions?

Reply:
We can make that call outside try. We get back a status which we validate within the try. I followed the same pattern as this - https://github.com/microsoft/onnxruntime/blob/e6da5946d1c7ccaef227f59d9357e74f9d649c1b/csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs#L810

Reply:
Also, we don’t really have to encapsulate the call in try as the native API implementation catches any exceptions and returns a status which we handle within the try catch block here.